lib/async-emitter.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const { iter } = require('@metarhia/common');
+
+class AsyncEmitter {
+  constructor() {
+    this.events = new Map();
+    this.wrappers = new Map();
+  }
+
+  on(name, fn) {
+    let event = this.events.get(name);
+    if (!event) {
+      event = new Set();
+      this.events.set(name, event);
+    }
+    event.add(fn);
+  }
+
+  once(name, fn) {
+    if (fn === undefined) {
+      return new Promise(resolve => {
+        this.once(name, resolve);
```

Comment:
This should also handle 'error' events, no?
i.e.
`this.once('error', reject)`

Reply:
It is not that simple. What if `name` is emitted, and then `error` is emitted.

Reply:
`ae.emit('error')` is a different event name, I think we shouldn't fire `await ae.on('e1')` if somebody calls `ae.emit('error')` @lundibundi